schema/test/unit/database_functions/grant_permissions_test.sql

```diff
@@ -0,0 +1,87 @@
+set client_min_messages to warning;
+create extension if not exists pgtap;
+reset client_min_messages;
+
+begin;
+select plan(9);
+
+create table ggircs_portal.test_table
+(
+  id integer primary key generated always as identity
+);
+
+create table ggircs_portal.test_table_specific_column_grants
+(
+  id integer primary key generated always as identity,
+  allowed text,
+  denied text
+);
+
+select has_function(
+  'ggircs_portal', 'grant_permissions',
+  'Function grant_permissions should exist'
+);
+
+select throws_ok(
+  $$
+    select ggircs_portal.grant_permissions('badoperation', null, 'test_table', 'ciip_administrator');
+  $$,
+  'P0001',
+  'invalid operation variable. Must be one of [select, insert, update, delete]',
+  'Function grant_permissions throws an exception if the operation variable is not in (select, insert, update, delete)'
+);
+
+select lives_ok(
+  $$
+    select ggircs_portal.grant_permissions('select', null, 'test_table', 'ciip_administrator');
+  $$,
+  'Function grants select'
+);
+
+select lives_ok(
+  $$
+    select ggircs_portal.grant_permissions('insert', null, 'test_table', 'ciip_administrator');
+  $$,
+  'Function grants insert'
+);
+
+select lives_ok(
+  $$
+    select ggircs_portal.grant_permissions('update', null, 'test_table', 'ciip_administrator');
+  $$,
+  'Function grants update'
+);
+
+select lives_ok(
+  $$
+    select ggircs_portal.grant_permissions('delete', null, 'test_table', 'ciip_administrator');
+  $$,
+  'Function grants delete'
+);
+
+select table_privs_are (
```

Comment:
we could probably do with a privs_are check at the beginning of this test, and between each `lives_ok`